android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java

```diff
@@ -46,12 +67,120 @@ public void setUp() {
     }
 
     @Test
-    public void justSomeInteraction() {
-        onView(withId(R.id.tabLayout)).check(matches(isDisplayed()));
-        onView(withId(R.id.store_text_layout)).check(matches(isDisplayed()));
-        onView(withId(R.id.store_text)).check(matches(isDisplayed()));
-        onView(withId(R.id.store_text)).perform(typeText("123"), closeSoftKeyboard());
-        onView(withId(R.id.category_text)).perform(click());
+    public void testBothPickers() {
+
+        onView(withId(R.id.date_layout))
+                .perform(AddingFragmentTest.clickIcon(true));
+
+        onView(withId(com.google.android.material.R.id.confirm_button))
+                .perform(click());
+
+        onView(withId(com.google.android.material.R.id.material_timepicker_ok_button))
+                .perform(click());
+
+    }
+
+    @Test
+    public void testDatePickerCancelTimePicker() {
+
+        onView(withId(R.id.date_layout))
+                .perform(AddingFragmentTest.clickIcon(true));
+
+        onView(withId(com.google.android.material.R.id.confirm_button))
+                .perform(click());
+
+        onView(withId(com.google.android.material.R.id.material_timepicker_cancel_button))
+                .perform(click());
+
+    }
+
+    @Test
+    public void testSetDatePickerCancelTimePicker() {
+
+
+        onView(withId(R.id.date_layout))
+                .perform(AddingFragmentTest.clickIcon(true));
+
+        onView(withId(com.google.android.material.R.id.mtrl_picker_header_toggle))
+                .perform(click());
+
+        onView(allOf(TestMatchers.childAtPosition(TestMatchers.childAtPosition(withId(com.google.android.material.R.id.mtrl_picker_text_input_date),
+                0), 0), isDisplayed()))
+                .perform(replaceText(date), closeSoftKeyboard());
+
+        onView(allOf(withId(com.google.android.material.R.id.confirm_button), isDisplayed()))
+                .perform(click());
+
+        onView(withId(com.google.android.material.R.id.material_timepicker_cancel_button))
+                .perform(click());
+
+        onView(withId(R.id.date_text)).check(matches(withText(date)));
+
+
+    }
+
+
+    @Test
+    public void testDatePickerTimePicker() {
+
+
+        onView(withId(R.id.date_layout))
+                .perform(AddingFragmentTest.clickIcon(true));
+
+        onView(withId(com.google.android.material.R.id.mtrl_picker_header_toggle))
+                .perform(click());
+
+        onView(allOf(TestMatchers.childAtPosition(TestMatchers.childAtPosition(withId(com.google.android.material.R.id.mtrl_picker_text_input_date),
+                0), 0), isDisplayed()))
+                .perform(replaceText(expectedDate), ViewActions.closeSoftKeyboard());
+
+        onView(allOf(withId(com.google.android.material.R.id.confirm_button), isDisplayed()))
+                .perform(click());
+
+        onView(allOf(withClassName(is("com.google.android.material.textfield.TextInputLayout")),withText(" ")))
```

Comment:
this espresso test still throws an error. I tried also only die withText Matcher but it didn't work

Reply:
```suggestion
        onView(withText(today))
```
should be sufficient. (you need to add the variable like you can see [here](https://github.com/Cebox82/FiBo/blob/36f820d0f5cb9c44f270c11e230e111263176b25/android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java#L293))

It fails because there is no view of the type TextInputLayout that has the text of the current date. Instead it is a TextInputEditText (or something like that). The allOf() checks if there is the text with the given class but it does not exists. Due to that it can't perform the replaceText action.